soc/riscv/wch_qingke_v3/common/soc_irq.S

```diff
@@ -0,0 +1,61 @@
+/*
```

Comment:
This file seems a clone of `soc/riscv/riscv-privileged/common/soc_irq.S`.
Can this directory be put into `soc/riscv/riscv-privileged/`?

Reply:
I'm afraid no because it uses a different implementation of interrupt controller (`arch_irq_enable`, `arch_irq_disable`, and so).

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/e7340282a7950d4cb071e77faafc249aaa569a8b/soc/riscv/riscv-privileged/common/soc_common_irq.c#L17C5-L38C1

Please add conditions here and integrate directories.

Reply:
Would you give me an advice on the naming of the Kconfig option? It also need a custom implementation of `arch_cpu_idle` (removing the line https://github.com/zephyrproject-rtos/zephyr/blob/bc43e890fd7a6035438a47b8853b6fe652472da2/soc/riscv/riscv-privileged/common/idle.c#L21).

Reply:
Solved by implementing the intc driver as a CLIC. Also introduced a `CONFIG_RISCV_WFI_ON_CPU_IDLE`, explained in message of commit 4a5c732245.